mobius-core/src/main/java/com/spotify/mobius/EffectRouterBuilderImpl.java

```diff
@@ -141,25 +166,32 @@ public void dispose() {}
     return new SafeConnectable<>(MergedConnectable.create(connectables));
   }
 
-  private static class SubtypeFilteringConnectable<I, J extends I, O> implements Connectable<I, O> {
-    private final Class<J> handledClass;
-    private final Connectable<J, O> delegate;
+  private class SubtypeFilteringConnectable<G extends F> implements Connectable<F, E> {
+    private final Class<G> handledClass;
+    private final Connectable<G, E> delegate;
 
-    SubtypeFilteringConnectable(Class<J> handledClass, Connectable<J, O> delegate) {
+    SubtypeFilteringConnectable(
+        Class<G> handledClass,
+        Connectable<G, E> delegate,
+        Function<Connectable<? extends F, E>, Consumer<Throwable>> errorHandler) {
```

Comment:
Couldn't we simplify this to just `Consumer<Throwable>`? It's not like the `Connectable` changes per consumer. This is in fact how the Rx implementation does it. It first applies the `Function`, then uses the created `Consumer` as the `doOnError` consumer for the effect handler instead of creating one every time.

Reply:
I just realized there's a bug in the RxJava implementation. If I add some transformers as effect handlers, then I add a fatal exception handler, it will not be used to handle the transformers that have been added already. It'll only apply for the new ones.
This is the code for Rx
```
return effects
    .ofType(effectClass)
    .compose(effectHandler)
    .doOnError(onErrorFunction.apply(effectHandler));
```
and this is what it should be if we want to make sure we always use the latest effect handler:
```
return effects
    .ofType(effectClass)
    .compose(effectHandler)
    .doOnError(new Consumer<Throwable>() {
      @Override
      public void accept(Throwable throwable) throws Exception {
        onErrorFunction.apply(effectHandler).accept(throwable);
      }
    });
```
Although, I still think the `onErrorFunction` won't change when we build the router so I think it's better if we deferred applying the `doOnError` to the `build` method before we pass it on to the `MobiusEffectRouter`. But that would mean that you would need to create a new `FatalErrorHandlingConnectable` that wraps the the `SubtypeFilteringConnectable` and a `Consumer<Throwable>` and that's what the router actually receives.

Reply:
Oh actually ignore me. I didn't notice that was a transformer so it is deferred :)

Reply:
There's even a test that validates that it works. :)